Source/bmalloc/bmalloc/Map.h

```diff
@@ -94,12 +110,31 @@ template<typename Key, typename Value, typename Hash> class Map {
     template<typename Predicate>
     Bucket& find(const Key& key, const Predicate& predicate)
     {
+        unsigned keysChecked = 0;
+        bool setFirstEmpty = false;
+        Bucket& firstEmptyBucket = m_table[0];
+
         for (unsigned h = Hash::hash(key); ; ++h) {
             unsigned i = h & m_tableMask;
 
             Bucket& bucket = m_table[i];
             if (predicate(bucket))
                 return bucket;
+            if (allowDeleting == AllowDeleting::DeletingAllowed) {
+                if (bucket.key)
+                    ++keysChecked;
+                else {
+                    if (!setFirstEmpty) {
+                        firstEmptyBucket = bucket;
+                        setFirstEmpty = true;
+                    }
+                    if (keysChecked >= m_keyCount) {
+                        if (setFirstEmpty)
+                            return firstEmptyBucket;
+                        return bucket;
+                    }
+                }
+            }
```

Comment:
Nit: As we talked about offline, this could be cleaned up slightly.

Reply:
Fixed.